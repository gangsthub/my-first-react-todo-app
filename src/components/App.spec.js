import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import App from './App'

const renderer = new ShallowRenderer()

jest.mock('./toDos/todos.state', () => ({
  todosLSKey: 'a',
  initialState: []
}))

describe('App smoke test', () => {
  it('matched expected output', () => {
    renderer.render(<App />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
