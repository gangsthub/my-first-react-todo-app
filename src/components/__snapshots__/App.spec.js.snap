// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App smoke test matched expected output 1`] = `
<React.Fragment>
  <div
    className="app flex column justify-content-between w90p sm:w60p ma-auto "
  >
    <Header
      headerTitle="ToDo App"
    />
    <main
      className="sm:flex justify-content-between flex-1 scroll-y"
    >
      <section
        className="pb-4 sm:w70p md:w80p xl:w80p flex column justify-content-between"
      >
        <TodosHeader />
        <TodoList
          onRemove={[Function]}
          onToggle={[Function]}
        />
        <div
          className="sm:order-0"
        >
          <AddTodo
            onAdd={[Function]}
          />
        </div>
      </section>
      <aside
        className="sm:w30p md:w20p xl:w20p scroll-y flex column justify-content-between"
      >
        <h2
          className="mb-0"
        >
          Logging
        </h2>
        <HistoryLogging />
      </aside>
    </main>
    <Footer
      footerText="Footer"
    />
  </div>
</React.Fragment>
`;
